nixos/modules/services/mail/tlsrpt.nix

```
@@ -0,0 +1,324 @@
+              type = types.str;
+              default = "0220";
+              description = ''
+                Permissions on the UNIX socket.
+              '';
+            };
+
+            log_level = mkOption {
+              type = types.enum [
+                "debug"
+                "info"
+                "warning"
+                "error"
+                "critical"
+              ];
+              default = "info";
+              description = ''
+                Level of log messages to emit.
+              '';
+            };
+          };
+        };
+        default = { };
+        description = ''
+          Flags from {manpage}`tlsrpt-collectd(1)` as key-value pairs.
+        '';
+      };
+
+      extraFlags = mkOption {
+        type = with types; listOf str;
+        default = [ ];
+        description = ''
+          List of extra flags to pass to the tlsrpt-reportd executable.
+
+          See {manpage}`tlsrpt-collectd(1)` for possible flags.
+        '';
+      };
+
+      configurePostfix = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether to modify the local Postfix service to grant access to the collectd socket.
+        '';
+      };
+    };
+
+    fetcher = {
+      settings = mkOption {
+        type = types.submodule {
+          freeformType = format.type;
+          options = {
+            storage = mkOption {
+              type = types.str;
+              default = config.services.tlsrpt.collectd.settings.storage;
+              defaultText = lib.literalExpression ''
+                config.services.tlsrpt.collectd.settings.storage
+              '';
+              description = ''
+                Path to the collectd sqlite database.
+              '';
+            };
+
+            log_level = mkOption {
+              type = types.enum [
+                "debug"
+                "info"
+                "warning"
+                "error"
+                "critical"
+              ];
+              default = "info";
+              description = ''
+                Level of log messages to emit.
+              '';
+            };
+          };
+        };
+        default = { };
+        description = ''
+          Flags from {manpage}`tlsrpt-fetcher(1)` as key-value pairs.
+        '';
+      };
+    };
+
+    reportd = {
+      settings = mkOption {
+        type = types.submodule {
+          freeformType = format.type;
+          options = {
+            dbname = mkOption {
+              type = types.str;
+              default = "/var/lib/tlsrpt/reportd.sqlite";
+              description = ''
+                Path to the sqlite database.
+              '';
+            };
+
+            fetchers = mkOption {
+              type = types.str;
+              default = lib.getExe' cfg.package "tlsrpt-fetcher";
+              defaultText = lib.literalExpression ''
+                lib.getExe' cfg.package "tlsrpt-fetcher"
+              '';
```

```suggestion
              defaultText = lib.literalExpression ''lib.getExe' config.services.tlsrpt.package "tlsrpt-fetcher"'';
```
I know from search.nuschtos.de that this gets rendered in the docs and manual as a mutliline string or types.lines which we are not using here.